### PR TITLE
feature: User now can add custom directories

### PR DESCRIPTION
**Pull request by PredatorCZ**
Helo.
Created new setting category, where user can specify custom directories.
Custom dirs are only used for patterns, pattern includes, python scripts, magics, yara rules, constants and encodings.
Main reason is that anyone might want to use multiple git repos or sources for such files.
Not implemented for macos, because it uses only one path, but can be implemented as well.

Also added ux change, where setting categories are sorted the way they were actually created instead of alphabetical order.

Tested on linux.

### CODE REVIEW

**Comment by WerWolv** [comment]
Thanks a lot! This looks great

File: plugins/builtin/source/lang/en_US.cpp [hunk 1]
```diff
@@ -693,7 +693,8 @@ namespace hex::plugin::builtin {
                     { "hex.builtin.provider.disk.sector_size", "Sector Size" },
                     { "hex.builtin.provider.disk.reload", "Reload" },
 
-                { "hex.builtin.layouts.default", "Default" }
+                { "hex.builtin.layouts.default", "Default" },
+                { "hex.builtin.setting.folders", "Folders" }
```
Comment (WerWolv) [inline plugins/builtin/source/lang/en_US.cpp:697]: This belongs further up (Line 677) below the other settings related strings I think

File: plugins/builtin/source/content/settings_entries.cpp [hunk 2]
```diff
@@ -219,6 +221,48 @@ namespace hex::plugin::builtin {
 
             return false;
         });
+
+        static const std::string dirsSetting { "hex.builtin.setting.folders" };
+
+        ContentRegistry::Settings::add(dirsSetting, dirsSetting, std::vector<std::string> {}, [](auto name, nlohmann::json &setting) {
```
Comment (WerWolv) [inline plugins/builtin/source/content/settings_entries.cpp:227]: This UI might benefit from a small help text explaining what it's actually doing.
Just a short text like "Adding folders here will cause ImHex to consider them as search paths for patterns, includes, magic files, etc."

File: lib/libimhex/source/api/content_registry.cpp [hunk 3]
```diff
@@ -162,6 +188,15 @@ namespace hex {
             return settings;
         }
 
+        std::vector<std::string> getStringVector(const std::string &unlocalizedCategory, const std::string &unlocalizedName) {
```
Comment (WerWolv) [inline lib/libimhex/source/api/content_registry.cpp:191]: I'd call this `getStringArray` because JSON refers to it as array

**Review by WerWolv — CHANGES_REQUESTED** [review]

File: plugins/builtin/source/content/settings_entries.cpp (hunk 2 above)
Comment (PredatorCZ) [reply plugins/builtin/source/content/settings_entries.cpp:229]: I added new methods for adding descriptions for categories.
They override header line that is under tabs, so the description doesn't waste space.

**Review by WerWolv — APPROVED** [review]

File: plugins/builtin/source/content/settings_entries.cpp (hunk 2 above)
Comment (WerWolv) [reply plugins/builtin/source/content/settings_entries.cpp:229]: Thank you!

**Pull request merged**